kw6/reader.py

```diff
@@ -20,61 +20,118 @@ class Reader(BaseModel):
         from pathlib import Path
         import kw6
 
-        path = Path('...')
+        path = Path("...")
 
         for position in kw6.Reader.from_path(path):
             for camera in position.cameras:
                 camera.image.save(
-                    f'{position.header.frame_index}_{camera.header.camera_index}.png'
+                    f"{position.header.frame_index}_{camera.header.camera_index}.png"
                 )
-    '''
+    """
 
+    path: Path
     stream: io.BufferedReader
     cached_positions: Dict[int, int]
-    n_bytes_position: int
     initial_frame_index: int
     inconsistent_position_bytes: bool = False
 
     class Config:
         arbitrary_types_allowed = True
 
     @staticmethod
-    def from_path(path, header_path=None):
-        stream = Path(path).open('rb')
+    @validate_arguments
+    def from_path(path: Path, header_path: Path = None):
+        stream = io.BufferedReader(path.open("rb"))
```

Comment:
Doesn't `path.open("rb")` always return a `BufferedReader`?

Reply:
My bad, I thought I got an error message in `__iter__` because of it and thought it was just pydantic converting for us